### DISCUSSION TITLE
Supported components: "Rotary switches"?

### DISCUSSION THREAD

**Discussion by stllvdv** [shanteacontrols/OpenDeck#211] [Q&A] [answered]
Hey @paradajz et al.,

Wondering if rotary switches are supported? Use-case would be selecting from e.g. maybe 5 or 6 "Range" or "Waveform" positions like LO, 32, 16, 8, 4, 2 or Sine, Try, Square, etc…

Maybe something… [like this?](https://www.digikey.de/de/products/detail/c-k/A10605RNZQK/3751750)

Warm regards.

**Comment by paradajz** [comment, marked as answer]
They are supported only if you think of them in terms of buttons - because that's what they are. So if you have rotary switch with eg. 5 positions, you'd need to setup 5 buttons in the OpenDeck UI.

**Reply by stllvdv** [reply]
@paradajz Such a design as attached isn't supported, unfortunately, correct?

<img width="817" alt="rotary-switch" src="https://github.com/shanteacontrols/OpenDeck/assets/29223372/38178eaa-55d6-4f34-a023-41e1d2dadcf6">

**Reply by paradajz** [reply]
It's not.

**Reply by Wapata** [reply]
Hi !
Wired that way it's a potentiometer... With values that will always jump from step to step.
If your "Range" can be assigned to a CC value and the resistors interval is even (like in the picture, they are all the same) it should be good.
When I did that for an RC emitter, i did not put a capacitor nor the last resistor. I guess they are here for filtering the "jump". If the botom resistor prevent you to have a "zero" value, you can offset that in the OpenDeck interface.

**Reply by stllvdv** [reply]
> Wired that way it's a potentiometer..

Exactly! And works as expected for "Range" or "Waveform".

**Reply by eurorackworld** [reply]
wow this is a very good idea, not thought of this. Just to check, so one of you tried this setup and it worked in Opendeck right?

**Reply by Wapata** [reply]
I haven't 
But in Opendeck it should, the real question is on the other end, on the DAW or plugin.

**Reply by stllvdv** [reply]
I'm not sure it would be a perfect solution for every use-case, but it works as expected for what I intend to use it for, controlling "Range" and "Waveform", reliably generating values 0 24 51 78 105 127. Saves lots of cables for me…

**Reply by Wapata** [reply]
Nice ! So did you put the last capacitor and resistor in the end ?
ps: there is a lot of rotary switches, with a lot of position quantities, and even with mechanical-movable end stop. So lot of fun for making a Synthetiser panel

**Reply by stllvdv** [reply]
Built it exactly as above.
